contracts/PetPark.sol

```diff
@@ -4,4 +4,93 @@ pragma solidity ^0.8.0;
 
 contract PetPark {
 
+    address  contractOwner;
+    enum AnimalType {
+        NONE, FISH, CAT, DOG, RABBIT, PARROT
+    }
+    enum Gender{
+        MALE, FEMALE
+    }
+    struct Animal {
+        AnimalType animalType;
+        uint count;
+    }
+
+    struct Borrower {
+        Gender gender;
+        uint age;
+        uint petCount;
+        bool exists;
+        AnimalType animalType;
+    }
+
+    event Added(AnimalType animalType, uint animalCount);
+    event Borrowed(AnimalType animalType);
+    event Returned(AnimalType animalType);
+
+    mapping(AnimalType => Animal) animals;
```

Comment:
Is there a need to define a struct for Animal? Can't you make it mapping from animal type to number of animals?

The optimisation here is that you won't be storing animal type twice - once as key of mapping and another time in struct